velox/functions/prestosql/tests/JsonCastTest.cpp

```diff
@@ -366,6 +366,35 @@ TEST_F(JsonCastTest, fromDate) {
       {std::nullopt, std::nullopt, std::nullopt, std::nullopt});
 }
 
+TEST_F(JsonCastTest, fromDecimal) {
+  testCastToJson<int64_t>(
+      DECIMAL(9, 2),
+      {123456789, -333333333, 0, 5, -9, std::nullopt},
+      {"1234567.89"_sv,
+       "-3333333.33"_sv,
+       "0.00"_sv,
+       "0.05"_sv,
+       "-0.09"_sv,
+       std::nullopt});
+  // Cannot cast long DECIMAL to JSON currently
+  VELOX_ASSERT_THROW(
+      testCastToJson<int128_t>(
+          DECIMAL(38, 5),
+          {DecimalUtil::kLongDecimalMin,
+           0,
+           DecimalUtil::kLongDecimalMax,
+           HugeInt::build(0xFFFFFFFFFFFFFFFFull, 0xFFFFFFFFFFFFFFFFull),
+           HugeInt::build(0xffff, 0xffffffffffffffff),
+           std::nullopt},
+          {"-999999999999999999999999999999999.99999",
+           "0.00000",
+           "999999999999999999999999999999999.99999",
+           "-0.00001",
+           "12089258196146291747.06175",
+           std::nullopt}),
+      "Cannot cast DECIMAL(38, 5) to JSON");
```

Comment:
Currently, Velox does not support converting long decimal to JSON type, only short decimal to JSON type. In order to avoid ambiguity, I will modify the title of PR, stating that the change only applies to the short decimal. Thank you for your code view and comments.